src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/CreateReferralHelper.java

```diff
@@ -0,0 +1,162 @@
+        referralType.setIsUrgent(caseData.getIsUrgent());
+        referralType.setReferralSubject(caseData.getReferralSubject());
+        referralType.setReferralSubjectSpecify(caseData.getReferralSubjectSpecify());
+        referralType.setReferralDetails(caseData.getReferralDetails());
+        referralType.setReferralDocument(caseData.getReferralDocument());
+        referralType.setReferralInstruction(caseData.getReferralInstruction());
+        referralType.setReferentEmail(caseData.getReferentEmail());
+
+        referralType.setReferralDate(Helper.getCurrentDate());
+
+        UserDetails userDetails = userService.getUserDetails(userToken);
+        referralType.setReferredBy(userDetails.getFirstName() + " " + userDetails.getLastName());
+
+        referralType.setReferralStatus(ReferralStatus.AWAITING_INSTRUCTIONS);
+
+        referralType.setReferralHearingDate(getNearestHearingToReferral(caseData));
+
+        ReferralTypeItem referralTypeItem = new ReferralTypeItem();
+        referralTypeItem.setId(UUID.randomUUID().toString());
+        referralTypeItem.setValue(referralType);
+
+        List<ReferralTypeItem> referralCollection = caseData.getReferralCollection();
+        referralCollection.add(referralTypeItem);
+        caseData.setReferralCollection(referralCollection);
+        clearReferralDataFromCaseData(caseData);
+    }
+
+    /**
+     * Gets the next hearing date from the referral, returns "None" if no suitable hearing date exists.
+     * @param caseData contains all the case data
+     * @return Returns next hearing date in "dd MMM yyyy" format or "None"
+     */
+    private String getNearestHearingToReferral(CaseData caseData) {
+        List<HearingTypeItem> hearingCollection = caseData.getHearingCollection();
+
+        if (CollectionUtils.isEmpty(hearingCollection)) {
+            return "None";
+        }
+
+        Date nextHearingAfterReferral = null;
+        for (HearingTypeItem hearing : hearingCollection) {
+            String hearingDateString =
+                hearing.getValue().getHearingDateCollection().get(0).getValue().getHearingTimingStart();
+
+            if (hearingDateString != null && !hearingDateString.isEmpty()) {
+                Date hearingStartDate;
+                try {
+                    hearingStartDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS").parse(hearingDateString);
+                } catch (ParseException e) {
+                    log.info("Failed to parse hearing date when creating new referral");
+                    continue;
+                }
+
+                if (hearingStartDate.after(new Date())
+                    && (nextHearingAfterReferral == null || hearingStartDate.before(nextHearingAfterReferral))
+                ) {
+                    nextHearingAfterReferral = hearingStartDate;
+                }
+            }
+        }
+        return nextHearingAfterReferral == null ? "None" :

```

Comment:
strange how checkstyle didn't complain about the ? and : being on new lines

Reply:
That is odd, it should want it on a new line, could you move it down anyway just in case checkstyle starts complaining later?